wtxmgr/tx.go

```diff
@@ -574,6 +576,75 @@ func (s *Store) PruneUnconfirmed(height int32, stakeDiff int64) error {
 	return nil
 }
 
+// fetchAccountForPkScript fetches an account for a given pkScript given a
+// credit value, the script, and an account lookup function. It does this
+// to maintain compatibility with older versions of the database.
+func (s *Store) fetchAccountForPkScript(credVal []byte, unminedCredVal []byte,
+	pkScript []byte,
+	acctLookupFunc func(dcrutil.Address) (uint32, error)) (uint32, error) {
+	// Attempt to get the account from the mined credit. If the
+	// account was never stored, we can ignore the error and
+	// fall through to do the lookup with the acctLookupFunc.
+	if credVal != nil {
+		acct, err := fetchRawCreditAccount(credVal)
+		if err != nil {
+			storeErr, ok := err.(Error)
+			if !ok {
+				return 0, err
+			}
+			switch storeErr.Code {
+			case ErrNoExists:
+			case ErrData:
+			default:
+				return 0, err
+			}
+		}
+
+		if err == nil {
+			return acct, nil
+		}
+	}
+	if unminedCredVal != nil {
+		acct, err := fetchRawUnminedCreditAccount(unminedCredVal)
+		if err != nil {
+			storeErr, ok := err.(Error)
+			if !ok {
+				return 0, err
+			}
+			switch storeErr.Code {
+			case ErrNoExists:
+			case ErrData:
+			default:
+				return 0, err
+			}
+		}
+
+		if err == nil {
+			return acct, nil
+		}
+	}
+
+	// Neither credVal or unminedCredVal were passed, or if they
+	// were, they didn't have the account set. Figure out the
+	// account from the pkScript the expensive way.
+	_, addrs, _, err :=
+		txscript.ExtractPkScriptAddrs(txscript.DefaultScriptVersion,
+			pkScript, s.chainParams)
+	if err != nil {
+		return 0, err
+	}
+
+	// Only look at the first address returned. This does not
+	// handle multisignature or other custom pkScripts in the
+	// correct way, which requires multiple account tracking.
+	acct, err := acctLookupFunc(addrs[0])
+	if err != nil {
```

Comment:
I consider this stylistic, I strongly prefer to end all function with declared nil errors rather than returning a function.